mat64/dense.go

```diff
@@ -417,3 +420,25 @@ func (m *Dense) Augment(a, b Matrix) {
 	w := m.View(0, ac, br, bc).(*Dense)
 	w.Copy(b)
 }
+
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+func (m Dense) MarshalBinary() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	enc := binary.NewEncoder(buf)
```

Comment:
BinaryMarshaler and BinaryUnmarshaler should be the simplest thing, which is to take the matrix, and dump its data in binary form (int64, int64, float64, float64 ...). This is also possibly the most cross language encoding, as it is the simplest to encode and decode. 

We can support multiple kinds of encoding, for example we could add GobEncoder and VarintEncoder, but BinaryMarshaler should be the most basic version.